bamboo/unit_tests/test_unit_layer_channelwise_fully_connected_distconv.py

```diff
@@ -0,0 +1,334 @@
+import functools
+import operator
+import os
+import os.path
+import sys
+import numpy as np
+import pytest 
+
+# Bamboo utilities
+current_file = os.path.realpath(__file__)
+current_dir = os.path.dirname(current_file)
+sys.path.insert(0, os.path.join(os.path.dirname(current_dir), 'common_python'))
+import tools
+
+# ==============================================
+# Objects for Python data reader
+# ==============================================
+# Note: The Python data reader imports this file as a module and calls
+# the functions below to ingest data.
+
+# Data
+np.random.seed(20200113)
+_num_samples = 17
+_sample_dims = (8, 1, 2)
+_sample_size = functools.reduce(operator.mul, _sample_dims)
+_samples = np.random.normal(size=(_num_samples, _sample_size)).astype(np.float32)
+_bias = np.random.normal(loc=0, size=(_sample_dims[0], 1, 1)).astype(np.float32)
+
+# Sample access functions
+def get_sample(index):
+    return _samples[index, :]
+
+
+def num_samples():
+    return _num_samples
+
+
+def sample_dims():
+    return (_sample_size,)
+
+
+# ==============================================
+# Setup LBANN experiment
+# ==============================================
+
+def setup_experiment(lbann):
+    """Construct LBANN experiment.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+    mini_batch_size = num_samples() // 2
+    trainer = lbann.Trainer(mini_batch_size)
+    model = construct_model(lbann)
+    data_reader = construct_data_reader(lbann)
+    optimizer = lbann.NoOptimizer()
+    return trainer, model, data_reader, optimizer
+
+
+def create_parallel_strategy(num_channel_groups):
+    return {"channel_groups": num_channel_groups,
+            "filter_groups": num_channel_groups}
+
+
+def construct_model(lbann):
+    """Construct LBANN model.
+
+    Args:
+        lbann (module): Module for LBANN Python frontend
+
+    """
+
+    # Input data
+    # Note: Sum with a weights layer so that gradient checking will
+    # verify that error signals are correct.
+    x_weights = lbann.Weights(optimizer=lbann.SGD(),
+                              initializer=lbann.ConstantInitializer(value=0.0),
+                              name='input_weights')
+    x0 = lbann.WeightsLayer(weights=x_weights,
+                            dims=tools.str_list(_sample_dims))
+    x1 = lbann.Reshape(lbann.Input(), dims=tools.str_list(_sample_dims), name="Input_layer")
+    x = lbann.Sum(x0, x1, name="Adding_weight_layer")
+    x_lbann = x
+
+    # Objects for LBANN model
+    obj = []
+    metrics = []
+    callbacks = []
+
+    num_height_groups = tools.gpus_per_node(lbann)
```

Comment:
Let's make this name more accurate:
```suggestion
    num_channel_groups = tools.gpus_per_node(lbann)
```